src/main/java/com/icthh/xm/uaa/security/oauth2/tfa/TfaOtpAuthenticationEmbedded.java

```diff
@@ -0,0 +1,167 @@
+package com.icthh.xm.uaa.security.oauth2.tfa;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.InternalAuthenticationServiceException;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.util.Assert;
+
+import java.util.Collection;
+
+/**
+ * The {@link TfaOtpAuthenticationEmbedded} class.
+ */
+@Slf4j
+public class TfaOtpAuthenticationEmbedded extends TfaOtpAuthenticationProvider {
```

Comment:
It looks line this class and `TfaOtpAuthenticationOtpMs` are quite similar. Is there is a way to extract common logic to super class to avoid duplication?